main.go

```diff
@@ -238,74 +318,124 @@ func main() {
 		os.Exit(0)
 	}
 
-	if *logFile != "" {
-		f, err := os.OpenFile(*logFile, os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0600)
-		if err != nil {
-			log.Fatalf("Error opening logfile: %v", err)
-		}
-		defer f.Close()
+	// print version on start
+	log.WithField("version", VERSION).
+		Debug("starting smtprelay")
 
-		log.SetOutput(io.MultiWriter(os.Stdout, f))
-	}
+	go handleMetrics()
```

Comment:
Isn't there actually a race condition here? `handleMetrics` calls `registerMetrics`, but until that completes,  `requestsCounter`, `errorsCounter`, and `durationHistogram` are all `nil`. Since `handleMetrics` was invoked via goroutine, it's possible (albeit unlikely) that something in another goroutine could access them before `registerMetrics` finishes.

So I think you need to either remove the `registerMetrics()` call from `handleMetrics()` and call it here:
```suggestion
	registerMetrics()
	go handleMetrics()
```

Or move the `go` statement into `handleMetrics`.

Reply:
Hhmm, you may be right. We'll have to have a look at this